src/main/java/org/spongepowered/common/command/SpongeHelpCommand.java

```diff
@@ -0,0 +1,151 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.command;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Iterables;
+import org.spongepowered.api.service.pagination.PaginationBuilder;
+import org.spongepowered.api.service.pagination.PaginationService;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.TextBuilder;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.api.text.action.TextActions;
+import org.spongepowered.api.text.format.TextColors;
+import org.spongepowered.api.text.format.TextStyles;
+import org.spongepowered.api.util.command.CommandCallable;
+import org.spongepowered.api.util.command.CommandException;
+import org.spongepowered.api.util.command.CommandMapping;
+import org.spongepowered.api.util.command.CommandResult;
+import org.spongepowered.api.util.command.CommandSource;
+import org.spongepowered.common.Sponge;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.TreeSet;
+
+public class SpongeHelpCommand implements CommandCallable {
+
+    private static final Comparator<CommandMapping> COMMAND_COMPARATOR = new Comparator<CommandMapping>() {
+
+        @Override
+        public int compare(CommandMapping o1, CommandMapping o2) {
+            return o1.getPrimaryAlias().compareTo(o2.getPrimaryAlias());
+        }
+    };
+
+    @Override
+    public CommandResult process(final CommandSource source, String arguments) throws CommandException {
+        if (arguments.split(" ").length > 1) {
+            throw new CommandException(Texts.of("Usage: /help ", getUsage(source)));
+        }
+
+        if (Sponge.getGame().getCommandDispatcher().containsAlias(arguments)) {
+            source.sendMessage(getDescription(source, Sponge.getGame().getCommandDispatcher().get(arguments).get()));
+            return CommandResult.success();
+        }
+
+        int page = 1;
+        if (!arguments.isEmpty()) {
+            try {
+                page = Integer.parseInt(arguments);
```

Comment:
@zml2008
Shall the user type `/help` and then `/page 5` instead of just `/help 5`?
Im pretty sure a lot of plugins will offer a 'fix' for this unexpected behaviour. Especially for console like interfaces without click support.